### PR TITLE
Splitup ephemeral processing

### PR DESCRIPTION
**Pull request by anikaweinmann**
Splitup functions/methods in ephemeral_processing.py (part 1) (see #189)

### CODE REVIEW

File: src/actinia_core/resources/ephemeral_processing.py [hunk 1]
```diff
@@ -1109,24 +1126,33 @@ def _check_reset_region(self):
...
+        ew_res = ew_res * fak
+        errorid, stdout_buff, stderr_buff = self.ginit.run_module(
+                "g.region", ["nsres=%f" % ns_res, "ewres=%f" % ew_res, "-g"])
+        self.message_logger.info(stdout_buff)
+        if errorid != 0:
+            raise AsyncProcessError(
                     "Unable to adjust the region settings to nsres: "
                     "%f ewres: %f error: %s" % (ns_res, ew_res, stderr_buff))
-            raise AsyncProcessError(
+        raise AsyncProcessError(
```
Comment (griembauer) [inline src/actinia_core/resources/ephemeral_processing.py:1155]: Please check the indentation here

File: src/actinia_core/resources/ephemeral_processing.py [hunk 2]
```diff
@@ -1109,24 +1126,33 @@ def _check_reset_region(self):
...
                 "Region to large, set a coarser resolution to minimum nsres: "
```
Comment (griembauer) [inline src/actinia_core/resources/ephemeral_processing.py:1156]:
```suggestion
                "Region too large, set a coarser resolution to minimum nsres: "
```

**Review by griembauer — COMMENTED** [review]
Thank you, just some small suggestions!

File: src/actinia_core/resources/ephemeral_processing.py [hunk 3]
```diff
@@ -536,42 +534,9 @@ def _send_to_database(self, document, final=False):
...
-                            self.webhook_auth.split(':')[0],
-                            ':'.join(self.webhook_auth.split(':')[1:])))
-                else:
-                    r = requests.post(self.webhook_update,
-                                      json=json.dumps(response_model))
-                if r.status_code not in [200, 204]:
-                    raise AsyncProcessError(
-                        "Unable to access the update webhook URL %s"
-                        % self.webhook_update)
+                self._post_to_webhook(document, 'update')
```
Comment (mmacata) [inline src/actinia_core/resources/ephemeral_processing.py:539]: Great first split up! This makes totally sense. Hopefully we can discover more like these.

File: src/actinia_core/resources/ephemeral_processing.py [hunk 4]
```diff
@@ -956,6 +896,88 @@ def _create_temp_database(self, mapsets=None):
...
+                                mapsets_to_link.append((mapset_path, mapset))
+                        elif mapset not in mapsets_to_link and global_db is False:
+                            mapsets_to_link.append((mapset_path, mapset))
+                    else:
+                        raise AsyncProcessError(
+                            "Invalid mapset <%s> in location <%s>"
+                            % (mapset, self.location_name))
+        else:
+            raise AsyncProcessError(
+                "Unable to access global location <%s>" % self.location_name)
```
Comment (mmacata) [inline src/actinia_core/resources/ephemeral_processing.py:978]: "Unable to access global location.." Might be global or user.

File: src/actinia_core/resources/ephemeral_processing.py [hunk 5]
```diff
@@ -1109,25 +1131,34 @@ def _check_reset_region(self):
...
+        ns_res = ns_res * fak
+        ew_res = ew_res * fak
+        errorid, stdout_buff, stderr_buff = self.ginit.run_module(
+                "g.region", ["nsres=%f" % ns_res, "ewres=%f" % ew_res, "-g"])
+        self.message_logger.info(stdout_buff)
+        if errorid != 0:
+            raise AsyncProcessError(
                     "Unable to adjust the region settings to nsres: "
                     "%f ewres: %f error: %s" % (ns_res, ew_res, stderr_buff))
             raise AsyncProcessError(
```
Comment (mmacata) [inline src/actinia_core/resources/ephemeral_processing.py]: Are there two `AsyncProcessError` ? Or am I just not seeing some indentation or similar?

**Review by mmacata — CHANGES_REQUESTED** [review]
Great start for splitup! Largest file tackled!

File: src/actinia_core/resources/ephemeral_processing.py (hunk 5 above)
Comment (mmacata) [reply src/actinia_core/resources/ephemeral_processing.py]: Sorry, you already changed that.

**Pull request merged**